support-frontend/app/services/AuthenticationService.scala

```diff
@@ -1,38 +1,69 @@
 package services
 
-import com.gu.identity.cookie.IdentityKeys
-import com.gu.identity.play.AccessCredentials.{Cookies, Token}
-import com.gu.identity.play.AuthenticatedIdUser
-import com.gu.identity.play.AuthenticatedIdUser._
-import play.api.mvc.RequestHeader
+import cats.implicits._
+import com.gu.identity.auth.UserCredentials
+import com.gu.identity.model.User
+import com.gu.identity.play.IdentityPlayAuthService
+import config.Identity
+import org.http4s.Uri
+import play.api.mvc.{Cookie, RequestHeader}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class AuthenticationService(override val identityKeys: IdentityKeys) extends com.gu.identity.play.AuthenticationService {
-  override lazy val authenticatedIdUserProvider: Provider =
-    Cookies.authProvider(identityKeys).withDisplayNameProvider(Token.authProvider(identityKeys, "membership"))
+// The following classes were previously defined in identity-play-auth,
+// but have been removed as part of the changes to that library:
+// authenticating users via a call to identity API; removing periphery functionality.
+// They have been redefined here to reduce diff across PRs,
+// but these classes could get refactored / simplified / removed in subsequent PRs.
+sealed trait AccessCredentials
+object AccessCredentials {
+  case class Cookies(scGuU: String, guU: Option[String] = None) extends AccessCredentials {
+    val cookies: Seq[Cookie] = Seq(
+      Cookie(name = "SC_GU_U", scGuU)
+    ) ++ guU.map(c => Cookie(name = "GU_U", c))
+  }
+  case class Token(tokenText: String) extends AccessCredentials
 }
+case class IdMinimalUser(id: String, displayName: Option[String])
+case class AuthenticatedIdUser(credentials: AccessCredentials, user: IdMinimalUser)
 
 // TODO: consider porting this to identity-play-auth.
 class AsyncAuthenticationService(
-  authenticationService: AuthenticationService,
-  testUserService: TestUserService
+    identityPlayAuthService: IdentityPlayAuthService,
+    testUserService: TestUserService
 )(implicit ec: ExecutionContext) {
 
-  // TODO in follow-up PR:
-  // authenticate the user by making a call to identity API rather than using underyling authentication service.
-  def authenticateUser(requestHeader: RequestHeader): Future[Option[AuthenticatedIdUser]] =
-    Future.successful(authenticationService.authenticatedUserFor(requestHeader))
+  import AsyncAuthenticationService._
+
+  def authenticateUser(requestHeader: RequestHeader): Future[AuthenticatedIdUser] =
+    identityPlayAuthService.getUserFromRequest(requestHeader)
+      .map { case (credentials, user) => buildAuthenticatedUser(credentials, user) }
+      .unsafeToFuture()
+
+  def tryAuthenticateUser(requestHeader: RequestHeader): Future[Option[AuthenticatedIdUser]] =
+    authenticateUser(requestHeader)
+      .map(user => Option(user))
+      .handleError(_ => None) // TODO: log error?
```

Comment:
not sure, for refernce this is eventually used to decide whether to prefill the email address on contributions landing page here https://github.com/guardian/support-frontend/pull/1960/files#diff-04600d093ee81c82983e9c9f0a0ef971R115
Contributions will then immediately do another identity request at that point (getUser) (and also discards any error without specific logging)
We should probably be concerned if that ratio dropped significantly even if it's an expected situation,  My feeling is it maybe wouldn't hurt to log this at info level?  But it does raise the question, how would we know if this was broken.

Reply:
we should also avoid the two identity api requests in the long term, perhaps they can be consolidated in due course.  cc @tomrf1

Reply:
It's a good point re 'how would we know if it this was broken'. I've added an info level log and also a TODO  noting that we might want to change this in future.